app/controllers/entity_categories_controller.rb

```diff
@@ -0,0 +1,69 @@
+class EntityCategoriesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_group_category, only: %i[index new create edit update destroy]
+
+  def index
+    @group_category = GroupCategory.find(params[:group_category_id])
+    @entity_categories = @group_category.entity_categories
+  end
+
+  def new
+    @entity_category = @group_category.entity_categories.build
+    @group_categories = current_user.group_categories
+  end
+
+  def create
+    @entity_category = current_user.entity_categories.build(entity_category_params)
+    @group_categories = current_user.group_categories
+
+    if @entity_category.save
+      redirect_to group_category_entity_categories_path(@group_category),
+                  notice: 'Entity category was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # def edit
+  #   @entity_category = current_user.entity_categories.find(params[:id])
+  # end
+  def edit
+    @group_category = GroupCategory.find(params[:group_category_id])
+    @entity_category = current_user.entity_categories.find(params[:id])
+  end
+
+  def update
+    #   @entity_category = current_user.entity_categories.find(params[:id])
```

Comment:
- [ ] It would be nice if you remove the unused code to make your codebase clean and professional as suggested by the previous reviewer :+1: